ledger/ledger.go

```diff
@@ -40,73 +45,145 @@ type Ledger interface {
 	RootHash() string
 	// GetPreviousValue executes a get against a previous version of the ledger, using that version's root hash.
 	GetPreviousValue(previousRootHash, key string) (result string, err error)
+	// EraseRootHash re-claims any memory used by this version of history, preserving bits shared with other versions.
+	EraseRootHash(rootHash string) error
+	// Stats gives basic storage info regarding the ledger's underlying cache
+	Stats() cache.Stats
+	// GetAll returns the entire state of the ledger
+	GetAll() (map[string]string, error)
+	// GetAllPrevious returns the entire state of the ledger at an arbitrary version
+	GetAllPrevious(string) (map[string]string, error)
 }
 
 type smtLedger struct {
-	tree *smt
+	tree    *smt
+	history *history
+	// this map will grow forever.  how do we limit it?
+	keyCache byteCache
 }
 
 // Make returns a Ledger which will retain previous nodes after they are deleted.
-func Make(retention time.Duration) Ledger {
-	return smtLedger{tree: newSMT(hasher, nil, retention)}
+// the retention parameter has been removed in favor of EraseRootHash, but is left
+// here for backwards compatibility
+func Make(_ time.Duration) Ledger {
```

Comment:
So cache.NewTTL is not used now? I saw a few tests are still calling it, but in istio/istio we will not?

Reply:
It is used in newSMT, only because we don't have a simple cache to use...